orchestrator/gravity_utils/src/types/batches.rs

```diff
@@ -61,6 +64,68 @@ impl TransactionBatch {
             Token::Dynamic(fees),
         )
     }
+
+    /// this function displays info about this batch including metadata
+    /// such as the name of the ERC20, it's current value etc
+    pub async fn display_with_eth_info(&self, pubkey: EthAddress, web30: &Web3) {
+        let token = self.token_contract;
```

Comment:
This function does a lot of work for no reason if info logging is disabled. Can you add a check at the start of the function to exit early?

Reply:
thought about this when I was writing it, took a few minutes of research to figure out how and I've patched it in.